lua/lspconfig/leanls.lua

```diff
@@ -1,23 +1,36 @@
 local configs = require 'lspconfig/configs'
 local util = require 'lspconfig/util'
 
+local function find_toml(startpath)
+  return util.search_ancestors(util.path.dirname(startpath), function(path)
+    local toml_file = util.path.join(path, "leanpkg.toml")
+    if util.path.is_file(toml_file) then
+      return path
+    end
+  end)
+end
+
 configs.leanls = {
   default_config = {
-    cmd = {"lean-language-server", "--stdio"};
+    cmd = {"lean", "--server"};
     filetypes = {"lean"};
     root_dir = function(fname)
-      return util.find_git_ancestor(fname) or vim.loop.os_homedir()
+      return find_toml(fname) or util.find_git_ancestor(fname) or vim.loop.os_homedir()
```

Comment:
So instead of `find_toml`, you'd use `util.root_pattern("leanpkg.toml")(fname)`.

Reply:
I would also suggest using `util.path.dirname(fname)` instead of `vim.loop.os_homedir()`. The home directory usually isn't a good root directory. We've had issues with many language servers in the past caused by a fallback to the home directory.